test/cql-pytest/cassandra_tests/functions/operation_fcts_test.py

```diff
@@ -0,0 +1,819 @@
+
+        with original_column_names(cql) as cql:
+            assertColumnNames(execute(cql, table, "SELECT a % a, b % a, c % a, d % a, e % a, f % a, g % a, h % a FROM %s WHERE a = 1 AND b = 2 AND c = 23 % 5"),
+                          "a % a", "b % a", "c % a", "d % a", "e % a", "f % a", "g % a", "h % a")
+
+        assertRows(execute(cql, table, "SELECT a % a, b % a, c % a, d % a, e % a, f % a, g % a, h % a FROM %s WHERE a = 1 AND b = 2 AND c = 23 % 5"),
+                   row(0, 0, 0, 0, c_float(0.5).value, 0.5, 0, Decimal("0.5")))
+
+        assertRows(execute(cql, table, "SELECT a % b, b % b, c % b, d % b, e % b, f % b, g % b, h % b FROM %s WHERE a = 1 AND b = 2 AND c = 23 % 5"),
+                   row(1, 0, 1, 0, c_float(1.5).value, 0.5, 1, Decimal("0.5")))
+
+        assertRows(execute(cql, table, "SELECT a % c, b % c, c % c, d % c, e % c, f % c, g % c, h % c FROM %s WHERE a = 1 AND b = 2 AND c = 23 % 5"),
+                   row(1, 2, 0, 1, c_float(2.5).value, 0.5, 1, Decimal("2.5")))
+
+        assertRows(execute(cql, table, "SELECT a % d, b % d, c % d, d % d, e % d, f % d, g % d, h % d FROM %s WHERE a = 1 AND b = 2 AND c = 23 % 5"),
+                   row(1, 2, 3, 0, 1.5, 2.5, 3, Decimal("0.5")))
+
+        assertRows(execute(cql, table, "SELECT a % e, b % e, c % e, d % e, e % e, f % e, g % e, h % e FROM %s WHERE a = 1 AND b = 2 AND c = 23 % 5"),
+                   row(c_float(1.0).value, c_float(2.0).value, c_float(3.0).value, 4.0, c_float(0.0).value, 1.0, Decimal("1.5"), Decimal("3.0")))
+
+        assertRows(execute(cql, table, "SELECT a % f, b % f, c % f, d % f, e % f, f % f, g % f, h % f FROM %s WHERE a = 1 AND b = 2 AND c = 23 % 5"),
+                   row(1.0, 2.0, 3.0, 4.0, 5.5, 0.0, Decimal("0.5"), Decimal("2.0")))
+
+        assertRows(execute(cql, table, "SELECT a % g, b % g, c % g, d % g, e % g, f % g, g % g, h % g FROM %s WHERE a = 1 AND b = 2 AND c = 23 % 5"),
+                   row(1,2,3,4, Decimal("5.5"), Decimal("6.5"), 0, Decimal("1.5")))
+
+        assertRows(execute(cql, table, "SELECT a % h, b % h, c % h, d % h, e % h, f % h, g % h, h % h FROM %s WHERE a = 1 AND b = 2 AND c = 23 % 5"),
+                   row(Decimal("1.0"),
+                       Decimal("2.0"),
+                       Decimal("3.0"),
+                       Decimal("4.0"),
+                       Decimal("5.5"),
+                       Decimal("6.5"),
+                       Decimal("7"),
+                       Decimal("0.0")))
+
+        # Test negation
+
+        with original_column_names(cql) as cql:
+            assertColumnNames(execute(cql, table, "SELECT -a, -b, -c, -d, -e, -f, -g, -h FROM %s WHERE a = 1 AND b = 2"),
+                          "-a", "-b", "-c", "-d", "-e", "-f", "-g", "-h")
+
+        assertRows(execute(cql, table, "SELECT -a, -b, -c, -d, -e, -f, -g, -h FROM %s WHERE a = 1 AND b = 2"),
+                   row(-1, -2, -3, -4, c_float(-5.5).value, -6.5, -7, Decimal("-8.5")))
+
+        # Test with null
```

Comment:
nit: other comments have extra blank line below
nit2: the comment says "null", but we pass "None"

Reply:
As before, this is an original Cassandra comment. The "null" in it refers to CQL's null (which conveniently is also called null in Java), and indeed in Python you pass a "None" to get a null.

Blank lines - some of them come from the original Cassandra code, but some of them are my mistakes (I need to add all these "with create_table" stuff and sometimes I accidentally change whitespace.